packages/core/src/common/menu.ts

```diff
@@ -89,11 +93,14 @@ export class MenuModelRegistry {
         const parent = this.findGroup(groupPath);
         let groupNode = this.findSubMenu(parent, menuId);
         if (!groupNode) {
-            groupNode = new CompositeMenuNode(menuId, label);
+            groupNode = new CompositeMenuNode(menuId, label, options ? options.iconClass : undefined);
             return parent.addNode(groupNode);
         } else {
             if (!groupNode.label) {
                 groupNode.label = label;
+                if (options) {
```

Comment:
it should be handled independent from label, can it happen groupnode already has iconclass?

Reply:
I did like that because somehow there was submenus registered without labels (it was the case of the "Open With" I tested on"). But I'll change it to:
```ts
if (!groupNode.iconClass && options)
    groupNode.iconClass = options.iconClass
```

Reply:
Does it harm somehow if icon is set but label not? If so  the let's keep it as is.